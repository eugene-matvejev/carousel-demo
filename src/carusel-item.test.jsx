import React from 'react';
import CaruselItem from './carusel-item';
import { shallow } from 'enzyme';
 
describe('<CaruselItem/>', () => {
    const props = {
        src: '{{src}}',
        tags: '{{tags}}',
    };

    describe('render', () => {
        it('with required/default props', () => {
            const c = shallow(<CaruselItem {...props}/>);

            expect(c).toMatchSnapshot();
        });

        describe('with optional props', () => {
            [
                ['qaSelector', '{{qaSelector}}'],                
            ].forEach(([prop, v]) => {
                it(`::${prop} as "${v}"`, () => {
                    const c = shallow(<CaruselItem {...props} {...{[prop]: v}}/>);

                    expect(c).toMatchSnapshot();
                });
            });
        });
    });
});