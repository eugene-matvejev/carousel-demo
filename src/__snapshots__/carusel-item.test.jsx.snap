// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CaruselItem/> render with optional props ::qaSelector as "{{qaSelector}}" 1`] = `
<div
  className="carusel-item"
  data-cy="{{qaSelector}}"
>
  <div
    className="carusel-item__image"
    data-cy="{{qaSelector}}"
    style={
      Object {
        "backgroundImage": "url(\\"{{src}}\\")",
      }
    }
  />
  <span
    className="carusel-item__tags"
    data-cy="{{qaSelector}}tags"
  >
    {{tags}}
  </span>
</div>
`;

exports[`<CaruselItem/> render with required/default props 1`] = `
<div
  className="carusel-item"
  data-cy=""
>
  <div
    className="carusel-item__image"
    data-cy=""
    style={
      Object {
        "backgroundImage": "url(\\"{{src}}\\")",
      }
    }
  />
  <span
    className="carusel-item__tags"
    data-cy="tags"
  >
    {{tags}}
  </span>
</div>
`;
